relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -60,13 +60,49 @@ func NewCosmosChainProcessor(log *zap.Logger, provider *cosmos.CosmosProvider) *
 }
 
 const (
-	queryTimeout                = 5 * time.Second
-	blockResultsQueryTimeout    = 2 * time.Minute
+	// Timeout for initialization queries and light block query,
+	queryTimeout = 10 * time.Second
+
+	// Timeout for block results query. Longer timeout needed for large blocks.
+	blockResultsQueryTimeout = 2 * time.Minute
+
+	// Back off delay between latest height query retries
 	latestHeightQueryRetryDelay = 1 * time.Second
-	latestHeightQueryRetries    = 5
 
+	// Max retries for failed latest height queries
+	latestHeightQueryRetries = 5
+
+	// Initial query loop minimum duration.
+	// Rolling average block time will be used after startup.
 	defaultMinQueryLoopDuration = 1 * time.Second
-	inSyncNumBlocksThreshold    = 2
+
+	// How many blocks behind the latest to consider in sync with the chain.
+	inSyncNumBlocksThreshold = 2
+
+	// Fixed delay between block query retry attempts.
+	blockQueryRetryDelay = 200 * time.Millisecond
+
+	// With doubling the 10 second queryTimeout each time, this gives ~3 mins max timeout.
+	blockQueryRetries = 4
+
+	// Each new delta block time will affect the rolling average by 1/n of this.
+	blockAverageSmoothing = 10
+
+	// Delta block times will be excluded from rolling average if they exceed this.
+	maxConsideredDeltaBlockTimeMs = 15000
+
+	// Scenarios for how much time trim should be added to
+	// target ideal block query window.
+
+	// Time trim addition when a block query fails
+	queryFailureTimeTrimAdditionMs = 73
```

Comment:
I used prime numbers (to avoid fixed multiples) that would allow a steady backoff in the case of errors (up to a limit), and a smaller number for successes to fine tune the window when blocks are expected to be available. This allows it to find the ideal window within ~15 blocks and then remain there. Outlier scenarios like one-off blocks that take an unexpectedly long amount of time to become available will not derail the targeting.